operator/v1/types_machineconfiguration.go

```diff
@@ -158,3 +179,170 @@ const (
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='None') ? size(self) == 1 : true", message="None action can only be specified standalone, as it will override any other actions"
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicySSHKey is actions to take for any SSHKey change
+type NodeDisruptionPolicySSHKey struct {
+	// actions represents the series of commands to be executed on changes to any sshkey changes
+	// through MachineConfig objects. This is an atomic list, which will be validated by
+	// the MachineConfigOperator, with any conflicts reflecting as an error in the
+	// status. If validation is successful, the actions will be applied in the order
+	// they are set in the list. If there are other incoming changes to other MachineConfig
+	// entries in the same update that require a reboot, the reboot will supercede these actions.
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=10
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='Reboot') ? size(self) == 1 : true", message="Reboot action can only be specified standalone, as it will override any other actions"
+	// +kubebuilder:validation:XValidation:rule="self.exists(x, x.type=='None') ? size(self) == 1 : true", message="None action can only be specified standalone, as it will override any other actions"
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Reload' ? has(self.reload) : !has(self.reload)",message="Reload is required when type is reload, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Restart' ? has(self.restart) : !has(self.restart)",message="Restart is required when type is restart, and forbidden otherwise"
+// +union
+type NodeDisruptionPolicyAction struct {
+	// type represents the commands that will be carried out if this NodeDisruptionPolicyActionType is executed
+	// Valid value(s): Reboot, Drain, Reload, Restart, DaemonReload, None, Special
+	// reload/restart requires a corresponding service target specified in the reload/restart field.
+	// Other values require no further configuration
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NodeDisruptionPolicyActionType `json:"type"`
+	// reload specifies the service to reload, only valid if type is reload
+	// +optional
+	Reload *ReloadService `json:"reload,omitempty"`
+	// restart specifies the service to restart, only valid if type is restart
+	// +optional
+	Restart *RestartService `json:"restart,omitempty"`
+}
+
+// ReloadService allows the user to specify the services to be reloaded
+type ReloadService struct {
+	// serviceName is the full name (e.g. crio.service) of the service to be reloaded
```

Comment:
Since we have this same sort of field multiple times, the godoc should probably be similar, as above, would like to see examples and enumerate the valid suffixes ideally